### ISSUE TITLE
Jasmin can't find the property when the class extends Error

### ISSUE THREAD

**Issue by magiconch** [jasmine/jasmine#1806]
## Expected Behavior

expect: baseError.errorCode == 1

## Current Behavior
Error: Expected undefined to equal 1.

## Suite that reproduces the behavior (for bugs)
```js
describe("Base Error test", () => {
    beforeEach(() => {
        var __extends = (this && this.__extends) || (function () {
            var extendStatics = function (d, b) {
                extendStatics = Object.setPrototypeOf ||
                    ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
                    function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
                return extendStatics(d, b);
            };
            return function (d, b) {
                extendStatics(d, b);
                function __() { this.constructor = d; }
                d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
            };
        })();
        var BaseError = /** @class */ (function (_super) {
            __extends(BaseError, _super);
            function BaseError() {
                var _this = _super.call(this) || this;
                _this.__errorCode = 1;
                return _this;
            }
            Object.defineProperty(BaseError.prototype, "errorCode", {
                get: function () {
                    return this.__errorCode;
                },
                set: function (value) {
                    this.__errorCode = value;
                },
                enumerable: true,
                configurable: true
            });
            return BaseError;
        }(Error));
        
        baseError = new BaseError();
    })
    it("nothing", () => {
        expect(baseError.errorCode).toEqual(1);
    });
});
```

## Your Environment

* Version used: 3.5.0
* Environment name and version: node.js: v11.8.0
* Operating System and version (desktop or mobile): macOS
* Link to your project: [demo](https://github.com/magiconch/jasmine_Error_Demo)

**Comment by sgravrock** [comment]
Thanks for providing a runable repro.

When I copied the code from the `beforeEach` and ran it in Node, I found that the value of `baseError.errorCode` is `undefined`. So in this case the matcher is correctly indicating a problem in the code under test. I didn't dig into it very deeply but it looks like the `BaseError` constructor isn't returning instances of `BaseError`:

```
> baseError.errorCode
undefined
> baseError.__proto__
Error {}
> BaseError.prototype
[BaseError] {
  constructor: [Function: BaseError],
  errorCode: [Getter/Setter]
}
```

If the inheritance were set up correctly, `baseError.__proto__` and `BaseError.prototype` would be the same object, which would result in `baseError` having an `errorCode` property.
